### ISSUE TITLE
support static fields in treemath.struct

### ISSUE THREAD

**Issue by GallagherCommaJack** [google/tree-math#21]
it would be nice to have some fields be pytree nodes and others not, which would make this a full replacement for `flax.struct`

**Comment by shoyer** [comment]
Yes, this would be nice. I would happily accept a PR adding this feature.

**Comment by Riccardo231** [comment]
Hello, I'd like to work on the project by fixing this issue. Can you assign this to me? Thanks

**Comment by shoyer** [comment]
Feel free to send in a PR!

**Comment by Riccardo231** [comment]
Good evening. Just for clarification, I don't understand well what the issue wants. I thought of making pytree_values and non_pytree_values arrays in the tree_flatten and tree_unflatten function. Am I right or am I missing something? Thank you
